lib/dockerregistry/inventory.go

```diff
@@ -2503,18 +2495,11 @@ func (sc *SyncContext) ClearRepository(
 		for req := range reqs {
 			reqRes := RequestResult{Context: req}
 			jsons, errors := getJSONSFromProcess(req)
-			if len(errors) > 0 {
-				reqRes.Errors = errors
-				requestResults <- reqRes
-				// Don't skip over this request, because stderr here is ignorable.
-				// continue
-			}
```

Comment:
In this portion of code, we are filling the requestResults channel with the same error twice. Now that the WaitGroup drain logic (_wg.Add(-1)_) has been moved to the requestsResults processor, this means that the WaitGroup was getting drained twice for each request. So this code block had to be removed in order to fix issues with the e2e tests where the WaitGroup was at a negative state.

Reply:
I think I get what you're saying. Here's a rewording of it: This commit moves the semaphore reduction line to ExecRequests. If we do not remove this particular block of code, then it can be the case that we push 2x into requestResults, *for the same request*. This meant WaitGroup would subtract "1" 2x even though we only added "1" 1x (negative state). Did I get it right?

Reply:
Yup, that's exactly right!